src/zephyr_coap_req.c

```diff
@@ -281,6 +282,26 @@ static int golioth_coap_req_reply_handler(struct golioth_coap_req *req,
     }
     else
     {
+        if (coap_get_option_int(response, COAP_OPTION_BLOCK1) >= 0)
+        {
+            /* This response has block1 */
+            if (coap_update_from_block(response, &req->block_ctx) == 0)
+            {
+                golioth_coap_request_msg_t *rmsg = req->user_data;
```

Comment:
We should not be modifying user data inside this function. The expectation of the caller is that user data is passed unmodified.

Rather trying to overload the user_data with more than one purpose, we should additional arguments to the functions in the call chain so we can pass back the negotiated block size explicitly.